lib/config/fileconf.go

```diff
@@ -535,6 +535,35 @@ type Auth struct {
 	// WebIdleTimeout sets global cluster default setting for WebUI client
 	// idle timeouts
 	WebIdleTimeout types.Duration `yaml:"web_idle_timeout,omitempty"`
+
+	// CAKeyParams configures how CA private keys will be created and stored.
+	CAKeyParams *CAKeyParams `yaml:"ca_key_params,omitempty"`
+}
+
+// CAKeyParams configures how CA private keys will be created and stored.
+type CAKeyParams struct {
+	// PKCS11 configures a PKCS#11 HSM to be used for private key generation and
+	// storage.
+	PKCS11 PKCS11 `yaml:"pkcs11,omitempty"`
+}
+
+// PKCS11 configures a PKCS#11 HSM to be used for private key generation and
+// storage.
+type PKCS11 struct {
+	// ModulePath is the path to the PKCS#11 library.
+	ModulePath string `yaml:"module_path"`
+	// TokenLabel is the CKA_LABEL of the HSM token to use. Set this or
+	// SlotNumber to select a token.
+	TokenLabel string `yaml:"token_label,omitempty"`
+	// SlotNumber is the slot number of the HSM token to use. Set this or
+	// TokenLabel to select a token.
+	SlotNumber *int `yaml:"slot_number,omitempty"`
+	// Pin is the raw pin for connecting to the HSM. Set this or PinPath to set
+	// the pin.
+	Pin string `yaml:"pin,omitempty"`
+	// PinPath is a path to a file containing a pin for connecting to the HSM.
+	// Set this or Pin to set the pin.
+	PinPath string `yaml:"pin_path,omitempty"`
```

Comment:
If these two are mutually-exclusive, is it checked anywhere (like, in some Check() method)?

Reply:
I now check this in `applyKeyStoreConfig`, the rest of the parameters are checked in `keystore.Config.CheckAndSetDefaults` but these don't make it that far